### PR TITLE
[api-draft][connector] Add MongoDB source connector

### LINKED ISSUE #2023: [api-draft][connector] Add MongoDB source connector
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Add MongoDB source connector

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wuchunfu**
## Purpose of this pull request

close #2023

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors/seatunnel-connectors-seatunnel/seatunnel-connector-seatunnel-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/source/MongodbSourceSplitEnumerator.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    public void handleSplitRequest(int subtaskId) {
+
+    }
+
+    @Override
+    public void registerReader(int subtaskId) {
+        readers.add(subtaskId);
+        if (assigned < 0) {
+            assigned = subtaskId;
+            context.assignSplit(subtaskId, new MongodbSourceSplit());
```
Comment (Hisoka-X) [inline seatunnel-connectors/seatunnel-connectors-seatunnel/seatunnel-connector-seatunnel-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/source/MongodbSourceSplitEnumerator.java:87]: If every reader will get same split, will it cause duplicate data to be read?

Comment (Hisoka-X) [reply seatunnel-connectors/seatunnel-connectors-seatunnel/seatunnel-connector-seatunnel-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/source/MongodbSourceSplitEnumerator.java:87]: Sorry I miss something.

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Comment by wuchunfu** [comment]
@CalvinKirs @ruanwenjun PTAL, thx.

File: tools/dependencies/known-dependencies-jdk11.txt [hunk 2]
```diff
@@ -507,7 +507,10 @@ metrics-core-3.2.1.jar
 metrics-core-4.0.0.jar
 minlog-1.3.0.jar
 mongo-java-driver-3.4.2.jar
+mongodb-driver-3.12.11.jar
+mongodb-driver-core-3.12.11.jar
 mongo-spark-connector_2.11-2.2.0.jar
```
Comment (CalvinKirs) [inline tools/dependencies/known-dependencies-jdk11.txt:512]: The newly added jar is not reflected in the license

Comment (CalvinKirs) [reply tools/dependencies/known-dependencies-jdk11.txt:512]: https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/release-docs/LICENSE

**Comment by wuchunfu** [comment]
@CalvinKirs @ruanwenjun @ashulin @Hisoka-X PTAL, thx.

**Comment by EricJoy2048** [comment]
Can you submit the document of this connector together in this PR?

**Comment by wuchunfu** [comment]
> Can you submit the document of this connector together in this PR?

@EricJoy2048 Ok, I will finish it later.

**Comment by CalvinKirs** [comment]
https://github.com/mongodb/mongo-java-driver/blob/3.12.x/THIRD-PARTY-NOTICES  please add this to `NOTICE` file

**Comment by CalvinKirs** [comment]
> https://github.com/mongodb/mongo-java-driver/blob/3.12.x/THIRD-PARTY-NOTICES please add this to `NOTICE` file

Usually, for the other party's NOTICE, we also need to keep it.

**Comment by EricJoy2048** [comment]
Is there any other way for us to complete the merger of this PR as soon as possible？

**Comment by EricJoy2048** [comment]
Hi, are there any new updates about this issue. Are there any difficulties that need the help of the community?

**Pull request closed without merging**